libs/web3/src/lib/ethereum-transaction-dialog/ethereum-transaction-dialog.tsx

```diff
@@ -0,0 +1,130 @@
+import { t } from '@vegaprotocol/react-helpers';
+import { Dialog, Icon, Intent, Loader } from '@vegaprotocol/ui-toolkit';
+import { isEthereumError } from '../ethereum-error';
+import type { EthTxState, TxError } from '../use-ethereum-transaction';
+import { EthTxStatus } from '../use-ethereum-transaction';
+import { ConfirmRow, TxRow, ConfirmationEventRow } from './dialog-rows';
+
+export interface EthereumTransactionDialogProps {
+  title: string;
+  onChange: (isOpen: boolean) => void;
+  transaction: EthTxState;
+  // Undefined means this dialog isn't expecting an additional event for a complete state, a boolean
+  // value means it is but hasn't been received yet
+  requiredConfirmations?: number;
+}
+
+export const EthereumTransactionDialog = ({
+  onChange,
+  title,
+  transaction,
+  requiredConfirmations = 1,
+}: EthereumTransactionDialogProps) => {
+  const { status, error, confirmations, txHash } = transaction;
+  return (
+    <Dialog
+      open={transaction.dialogOpen}
+      onChange={onChange}
+      size="small"
+      {...getWrapperProps(title, status)}
+    >
+      <TransactionContent
+        status={status}
+        error={error}
+        txHash={txHash}
+        confirmations={confirmations}
+        requiredConfirmations={requiredConfirmations}
+      />
+    </Dialog>
+  );
+};
+
+export const TransactionContent = ({
+  status,
+  error,
+  txHash,
+  confirmations,
+  requiredConfirmations = 1,
+}: {
+  status: EthTxStatus;
+  error: TxError | null;
+  txHash: string | null;
+  confirmations: number;
+  requiredConfirmations?: number;
+}) => {
+  if (status === EthTxStatus.Error) {
+    const classNames = 'break-all text-black dark:text-white';
+    if (isEthereumError(error)) {
+      return (
+        <p className={classNames}>
+          {t('Error')}: {error.reason}
+        </p>
+      );
+    }
+
+    if (error instanceof Error) {
+      return (
+        <p className={classNames}>
+          {t('Error')}: {error.message}
+        </p>
+      );
+    }
+
+    return (
+      <p className={classNames}>
+        {t('Error')}: {t('Unknown error')}
+      </p>
+    );
```

Comment:
```suggestion
    const errorMessage = (isEthereumError(error) && error.reason) || (error instanceof Error && error.message) || t('Unknown error'); 
    
    return (
        <p className={classNames}>
          {t('Error')}: {errorMessage}
        </p>
      );
```

Reply:
Good call on tidying this, though I find that condition a little confusing so I've just kept if statements to set an error message so the UI isn't duplicated.